custom_components/ha_washdata/suggestion_engine.py

```
@@ -20,16 +20,408 @@
+    return f" Excluded {total} mis-detected cycle(s): {parts}."
+
+
+# ─── Parameter interdependency reconciliation (Stage 5g) ──────────────────────
+# Suggestions are produced by several independent passes, so a value for one
+# parameter can silently contradict another (e.g. a start threshold below the
+# stop threshold, or an off_delay longer than the cycle-separation gap). This
+# pass takes the full suggestion set plus the current option values and nudges
+# any *suggested* value so the coupled invariants hold, recording why.
+
+
+def _num(value: Any) -> float | None:
```

Ruff (0.15.20)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)